apps/web/src/services/cost-explorer/store/cost-analysis-page-store.ts

```diff
@@ -104,10 +104,14 @@ export const useCostAnalysisPageStore = defineStore('cost-analysis-page', {
             return '';
         },
         isPeriodInvalid: (state) => {
-            if (state.granularity === GRANULARITY.DAILY) {
-                return dayjs.utc().diff(state.period?.start, 'year') > 1;
-            }
-            return dayjs.utc().diff(state.period?.start, 'year') > 3;
+            const checkPeriod = (limit:number):{isStartInvalid:boolean, isEndInvalid:boolean} => {
+                const isStartInvalid = dayjs.utc().diff(state.period?.start, 'month') >= limit;
+                const isEndInvalid = dayjs.utc().diff(state.period?.end, 'month') >= limit;
+                return { isStartInvalid, isEndInvalid };
+            };
+            const isGranularityDaily = state.granularity === GRANULARITY.DAILY;
+            const { isStartInvalid, isEndInvalid } = checkPeriod(isGranularityDaily ? 12 : 36);
```

Comment:
question: what is this number 12, 36 meaning?

Reply:
Oh.. I will change this value to constant. 
If the granularity is daily, we can look up 1 year's data. For other granularities, it can be 3 years.